src/components/navigation/link.ts

```diff
@@ -120,12 +124,16 @@ export class Link {
     });
 
     const link = this.resolveLink({ hasEntitlement, config });
+    const isLinkTargetInSameProject =
+      project &&
+      config.projectIdentifier &&
+      project === config.projectIdentifier;
 
     this.link = this.prefixDomain({
```

Comment:
Wondering it it would make sense to move this to the prefixDomain logic in terms of having all logic related to prefixing the domain in there

Reply:
i opened a branch here: https://github.com/smg-automotive/components-pkg/pull/1249

I find it slightly harder to read but from a responsibility standpoint it makes sense. Let me know what you think. I am fine either way

Reply:
It merge it so we have it in one place should we ever refactor it and don't need to touch multiple parts

Reply:
thanks for the review! waiting for Vy to test both projects before going ahead